### PR TITLE
add instance_id generated at install time

### PR DESCRIPTION
**Pull request by bmc-msft**
## Summary of the Pull Request

Adds a randomly generated UUID that is unique to each install that is included in telemetry.

## PR Checklist
* [x] Applies to work item: #234 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [x] Requires documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

This adds an `instance_id` that is generated at install time, and add it to the supervisor and agent configs to be used if telemetry is enabled.

NOTE: Including a per-install randomly generated UUID as an instance ID was approved during the initial internal privacy review.

## Validation Steps Performed

Standard integration tests

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/azure/containers.py [hunk 1]
```diff
@@ -129,11 +129,11 @@ def save_blob(
 
 def get_blob(
     container: str, name: str, account_id: Optional[str] = None
-) -> Optional[Any]:  # should be bytes
+) -> Optional[bytes]:
     service = get_blob_service(account_id)
     try:
         blob = service.get_blob_to_bytes(container, name).content
-        return blob
+        return cast(bytes, blob)
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/azure/containers.py:136]: :+1:

**Review by chkeita — APPROVED** [review]

File: src/api-service/__app__/onefuzzlib/azure/creds.py [hunk 2]
```diff
@@ -102,6 +102,16 @@ def get_instance_url() -> str:
     return "https://%s.azurewebsites.net" % get_instance_name()
 
 
+@cached
+def get_instance_id() -> UUID:
+    from .containers import get_blob
+
+    blob = get_blob("base-config", "instance-id", account_id=get_func_storage())
+    if blob is None:
+        raise Exception("missing instance_id")
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/azure/creds.py:111]: Should this be `instance-id` in the exception message? Thinking about greppability.

Comment (ranweiler) [reply src/api-service/__app__/onefuzzlib/azure/creds.py:111]: Oh, actually, is the `instance-id` string above just incorrect? Looks like we set the snake case version below.

File: src/deployment/deploy.py [hunk 3]
```diff
@@ -464,6 +465,29 @@ def create_eventgrid(self):
                 % json.dumps(result.as_dict(), indent=4, sort_keys=True),
             )
 
+    def add_instance_id(self):
+        logger.info("setting instance_id log export")
+
+        container_name = "base-config"
+        blob_name = "instance_id"
```
Comment (ranweiler) [inline src/deployment/deploy.py:472]: Should be changed here or above.

**Review by ranweiler — CHANGES_REQUESTED** [review]

**Review by ranweiler — APPROVED** [review]

**Pull request merged**